File: RickAndMortyApp/RickAndMorty/Front/Modules/Start/Login/Presenter/RickAndMortyAppStart/LoginPresenter.m
//
//  RickAndMortyAppStart/LoginPresenter.m
//  RickAndMortyApp
//
//  Created by Cristian Peña on 22/04/2024.
//  Copyright © 2024 RickAndMortyApp. All rights reserved.
//

#import "RickAndMortyAppStart/LoginPresenter.h"

#import "RickAndMortyAppStart/LoginViewInput.h"
#import "RickAndMortyAppStart/LoginInteractorInput.h"
#import "RickAndMortyAppStart/LoginRouterInput.h"

@implementation RickAndMortyAppStart/LoginPresenter

#pragma mark - Методы RickAndMortyAppStart/LoginModuleInput

- (void)configureModule {
    // Стартовая конфигурация модуля, не привязанная к состоянию view
}

#pragma mark - Методы RickAndMortyAppStart/LoginViewOutput

- (void)didTriggerViewReadyEvent {
	[self.view setupInitialState];
}

#pragma mark - Методы RickAndMortyAppStart/LoginInteractorOutput

@end

File: RickAndMortyApp/RickAndMortyApp/Front/Features/Start/Modules/Login/User Interface/Presenter/LoginPresenter.swift
//
//  LoginPresenter.swift
//  RickAndMortyApp
//
//  Created by Cristian Peña on 22/04/2024.
//  Copyright © 2024 RickAndMortyApp. All rights reserved.
//

import UIKit

class LoginPresenter {

    weak var view: LoginViewInput!
    var interactor: LoginInteractorInput!
    var router: LoginRouterInput!
}


// MARK: LoginModuleInput

extension LoginPresenter: LoginModuleInput {
    
}


// MARK: LoginViewOutput

extension LoginPresenter: LoginViewOutput {
    
    func viewIsReady() {}
    
    func didTapInformationButton() {
        view.showTooltip()
    }
    
    func didTapSendButton() {
        view.showLoading()
        interactor.login()
    }
}


// MARK: LoginInteractorOutput

extension LoginPresenter: LoginInteractorOutput {
    
    func didLogin() {
        
        // TODO: The connection will be worked on in a next PR
        DispatchQueue.main.asyncAfter(deadline: .now() + 1.5) {
            
            self.view.hideLoading()
            self.router.routeToHome()
        }
    }
    
    func didFailLogin() {}
}
